src/libs/store.js

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useReducer } from 'react';
+
+const initialState = {
+    isLoading: false,
+};
+const store = createContext(initialState);
+const { Provider } = store;
+
+const StateProvider = ( { children } ) => {
+  const [state, dispatch] = useReducer((state, action) => {
```

Comment:
This looks really good and it is very valid React Context code.

I do have a few comments though:
- If you notice, the `useReducer` hook leans towards creating a redux-like state management system which just increases complexity. 
  - I would honestly prefer using redux instead of `useReducer` because with this approach, you're bringing in the same amount of boilerplate as redux (actions + reducers) with none of the performance benefits of redux
- I would advise not using the `useReducer` hook unless completely necessary because it adds a lot more complexity to a solution that could be a lot simpler just using the `useState` hook. For example, you could have a boolean state variable for display loading instead.
- The standard for using the React context api is to have a directory named "contexts"
  - The standard for contexts is to not have "global state" like we do in redux. Meaning there shouldn't be one file that connects all the state together.
  - The standard is to have multiple context files. Each context file would be small in scope and only store data that is relevant. For example, having a separate AuthContext and a FetchDataContext.

Reply:
This is great. Thanks!
@slimTanjim , using different contexts is probably the answer to our question about how do you prevent the entire app from re-rendering when a component only cares about a subset of the context state.
@ahmednkhan24 I'm down for removing the useReducer. We'll look into how to use the context directly with `useState`. Thanks for the advice